libs/server/repository/core/application-services/src/lib/queries/get-pr-timeline/get-pr-timeline.handler.ts

```diff
@@ -0,0 +1,55 @@
+import { QueryHandler, IQueryHandler } from '@nestjs/cqrs';
+import { GetPrTimelineQuery } from './get-pr-timeline.query';
+import {
+  PrRepository,
+  prRepositoryFactoryToken,
+  RepositoryRepository
+} from '@pimp-my-pr/server/repository/core/domain-services';
+import { PrTimelineReadModel } from '@pimp-my-pr/server/repository/core/application-services';
+import {
+  getStepsCount,
+  getTimeLineHistory,
+  InvalidTimelineParametersException,
+  PrEntity,
+  traversePagesUntil
+} from '@pimp-my-pr/server/repository/core/domain';
+import { prTimelineModelFactory } from '../../read-models/factories/pr-timeline-model.factory';
+import { Inject } from '@nestjs/common';
+import { Platform } from '@pimp-my-pr/shared/domain';
+
+@QueryHandler(GetPrTimelineQuery)
+export class GetPrTimelineHandler implements IQueryHandler<GetPrTimelineQuery> {
+  constructor(
+    @Inject(prRepositoryFactoryToken)
+    private prRepositoryFactory: (platform: Platform) => PrRepository,
+    private repoRepository: RepositoryRepository
+  ) {}
+
+  async execute(query: GetPrTimelineQuery): Promise<PrTimelineReadModel> {
+    const { repositoryId, step, timelineFrom, timelineTo, token, platform, createdAfter } = query;
+    const { fullName } = await this.repoRepository.getById(repositoryId);
+    const prRepository = this.prRepositoryFactory(platform);
+
+    const prs = await traversePagesUntil<PrEntity>(
```

Comment:
I'm afraid that this will be soon moved from here, because we need to implement support for bitbucket and gitlab

Reply:
For now, Bitbucket and GitLab should also be supported by this feature. I would consider this function as a temporary solution before we start implementing prs caching,